src/ui/browserAction/popupPage.js

```diff
@@ -51,30 +52,36 @@ export class BrowserActionPopup extends LitElement {
     pageURL: { type: String },
     _siteContext: { type: Object },
     hasSiteContext: { type: Boolean },
+    _siteContexts: { type: Array },
   };
 
   constructor() {
     super();
     this.pageURL = null;
     this._siteContext = null;
-    browser.tabs.onUpdated.addListener(() => this.updatePage());
-    browser.tabs.onActivated.addListener(() => this.updatePage());
-    vpnController.state.subscribe((s) => (this.vpnState = s));
+    vpnController.state.subscribe((s) => {
+      this.vpnState = null;
+      /** @type {VPNState} */
+      this.vpnState = s;
+    });
     vpnController.servers.subscribe((s) => (this.servers = s));
+    proxyHandler.siteContexts.subscribe((s) => {
+      this._siteContexts = s;
+      console.log(this._siteContexts);
```

Comment:
Not sure if that is needed :)

Reply:
ha! no we probably don't want to do this.